qiskit/circuit/library/standard_gates/pauli.py

```diff
@@ -0,0 +1,88 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Simulator command to perform multiple pauli gates in a single pass
+"""
+import numpy as np
+
+from qiskit.circuit.quantumregister import QuantumRegister
+from qiskit.circuit.library.standard_gates.i import IGate
+from qiskit.circuit.library.standard_gates.x import XGate
+from qiskit.circuit.library.standard_gates.y import YGate
+from qiskit.circuit.library.standard_gates.z import ZGate
+
+from qiskit.circuit.gate import Gate
+from qiskit.circuit.exceptions import CircuitError
+
+
+class PauliGate(Gate):
+    r"""A multi-qubit Pauli gate.
+
+        This gate exists for optimization purposes for the
+        quantum statevector simulation, since applying multiple
+        pauli gates to different qubits at once can be done via
+        a single pass on the statevector.
+
+        The functionality is equivalent to applying
+        the pauli gates sequentially using standard Qiskit gates
+        """
+
+    def __init__(self, pauli_string="I"):
+        if isinstance(pauli_string, float):
+            pauli_string = "I"
```

Comment:
This is certainly a welcome behavior, but it clashes with many automatic tests which implicitly assume the parameters a gate can have consist of a vector of floats (and sometimes just assign random parameters based on this assumption). For now it's either that or modification of the testing framework.

Reply:
@kdk Is the testing framework easy to modify for this? Seems not ideal to have to add this to the class just for the automatic tests

Reply:
I agree, this should be easy to fix. Which tests are failing? There are tests to keep some expectations on standard gates (e.g. that their gate parameters align with their definition parameters, ...) but they each already have cases handling gates that have non-standard signatures. In fact, it looks like those have already been updated to handle `PauliGate`s.

Reply:
I added workaround in the test files; you might have a better idea how to circumvent this.

https://github.com/Qiskit/qiskit-terra/pull/5153/files#diff-d45d22ad301d9187912920617d10dfd53aa0c1a1c80fca08b1a02acf2552e65eR179

The most problematic case is what we have in `test_to_matrix` and `test_to_matrix_op`: here the params are initialized once `params = [0.1 * i for i in range(1, 11)]`, before being applied to each gate; the only difference is the computed number of actual parameters used from this list.

Since `PauliGate` expects `str` and not `float`, this crashes without direct override (as I added).